pkgs/development/libraries/sundials/default.nix

```diff
@@ -5,7 +5,9 @@
 , blas
 , lapack
 , gfortran
-, lapackSupport ? true }:
+, suitesparse
+, lapackSupport ? true
+, kluSupport ? true }:
```

Comment:
are the klu related packages likely to break the build?

Reply:
I don't think it is necessary to disable klu support in sundials for scikits-odes, since build+test is ok.

Reply:
Via some conditionals, it could be disabled only for `python2Packages.scikits-odes` :) but TBH I wouldn't see benefit in this effort - according to `git grep`, there are no reverse dependencies of `python2Packages.scikits-odes` available, and Python 2 is EOL (!!). `python3Packages.scikits-odes` builds fine.

> are the klu related packages likely to break the build?

What do you mean @jonringer?

Reply:
The biggest argument against adding "optional" dependencies by default is that it makes more fragile since there's another point of potential failure (either the dependency doesn't build, or the dependency gets bumped and is no longer compatible).

That's the concern I'm expressing. 

Unless the klu library is prone to breakages, we should probably add it.

Reply:
> either the dependency doesn't build

`suitesparse` is what provides KLU support for sundials (IDK what that even means). [Suitesparse](https://github.com/DrTimothyAldenDavis/SuiteSparse) seems well maintained to me, both upstream and [at Nixpkgs](https://github.com/NixOS/nixpkgs/blob/800d9e77fe80c98b93e60a33e803f10d477f0664/pkgs/development/libraries/science/math/suitesparse/default.nix#L60-L61).

> or the dependency gets bumped and is no longer compatible

The best we can do is to trust Sundials to test it with the current versions of suitesparse we ship. @GuillaumeDesforges perhaps it'll be nice to note in a comment near the conditional `kluSupport`, that this feature is tested by upstream according to the [INSTALL_GUIDE.pdf](https://raw.githubusercontent.com/LLNL/sundials/master/INSTALL_GUIDE.pdf), section 1.1.4. Currently they write there:

> The KLU libraries are part of SuiteSparse, a suite of sparse matrix software, available from the TexasA&M University website:http://faculty.cse.tamu.edu/davis/suitesparse.html.sundialshasbeen tested with SuiteSparse version 5.3.0

While we ship version 5.7.2 of suitesparse, so IDK :ok_woman:.

As a side note, I also support enabling this feature by default, for GNU Octave 6.0.0 which is about to be released. It will require this feature as well from this major version of sundials (5) . Currently Octave uses an old version of sundials with this feature enabled as well, see https://github.com/NixOS/nixpkgs/pull/79864/files#diff-a71f6e05947eac8bd8360ddb7faa9693 .

Reply:
I don't understand the issue. As with most nixpkgs packages, if sundials requires a bumped version of suitesparse we can just upgrade suitesparse, and if it breaks other packages we can create another bumped package like `suitesparse_X_X_X` (like in the Haskell infrastructure where multiple versions of a package live).

> Suitesparse seems well maintained to me, both upstream and at Nixpkgs.

Agreed, I would have been more concerned if it wasn't the case but it seems we are relatively safe here